### PR TITLE
fixes #22065; do not search path for relative imports

### LINKED ISSUE #22065: `import ./modulename` ignores the `./` path prefix
### Description

```nim
import "."/tables
import ./tables
```
even in a directory with no `tables.nim` or other way of finding a `tables` module` to import, compiles.

https://nim-lang.org/docs/manual.html#modules-import-statement doesn't seem to explicitly document what should happen in this situation, but "." is a way of specifying a path. Indeed, attempting to import `./std/tables`, for example, does not work, because it is being treated as a path, so there's an inconsistency here at least.

### Nim Version

```
Nim Compiler Version 1.9.3 [Linux: amd64]
Compiled at 2023-06-10
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: b2d77619757c73a8b14a8ccf127eaf3e630159cf
active boot switches: -d:release
```

```
Nim Compiler Version 1.6.12 [Linux: amd64]
Compiled at 2023-06-08
Copyright (c) 2006-2023 by Andreas Rumpf

git hash: 1aa9273640c0c51486cf3a7b67282fe58f360e91
active boot switches: -d:release
```

### Current Output

```text
/tmp/b.nim(2, 8) Hint: duplicate import of 'tables'; previous import here: /tmp/b.nim(1, 11) [DuplicateModuleImport]
/tmp/b.nim(1, 11) Warning: imported and not used: 'tables' [UnusedImport]
/tmp/b.nim(2, 8) Warning: imported and not used: 'tables' [UnusedImport]
```

### Expected Output

```text
Error: cannot open file: tables
```

### Possible Solution

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ringabout**
fixes #22065

### CODE REVIEW

File: compiler/options.nim [hunk 1]
```diff
@@ -923,7 +924,11 @@ proc findModule*(conf: ConfigRef; modulename, currentModule: string): AbsoluteFi
     else: # If prefixed with std/ why would we add the current module path!
       let currentPath = currentModule.splitFile.dir
       result = AbsoluteFile currentPath / m
-    if not fileExists(result):
+      if not fileExists(result) and m[0] == '.' and m[1] != '.':
```
Comment (metagn) [inline compiler/options.nim]: Any reason to exclude `..`?

Comment (ringabout) [reply compiler/options.nim]: Well, I was tired yesterday. It was just a placeholder and a draft per se.

**Comment by arnetheduck** [comment]
btw, something to be aware of is that this kind of "strictness" increase will result in a number of packages failing to compile - this is certainly the case of status' packages which prompted our initial desire for this feature - worst case, this might have to be turned into a warning ("imported from weird location") that can be turned into an error

**Comment by ringabout** [comment]
> something to be aware of is that this kind of "strictness" increase will result in a number of packages failing to compile - this is certainly the case of status' packages which prompted our initial desire for this feature - worst case, this might have to be turned into a warning ("imported from weird location") that can be turned into an error

As to important packages, only arraymancer uses `./math`. I will add a changelog. A compile error is good, no to mention, `./math` doesn't actually make sense, which can be though of a bug. Warnings + warnAsError might be a better solution, though.

File: changelogs/changelog_2_0_0.md [hunk 2]
```diff
@@ -260,6 +260,8 @@
 
 - `strutils.split` and `strutils.rsplit` now forbid an empty separator.
 
+- Relative imports will not resolve to stdlib anymore, e.g. `import ./tables` now reports an error properly.
```
Comment (arnetheduck) [inline changelogs/changelog_2_0_0.md:263]: wait .. only `stdlib` or also `-p:path`? ie the latter should also not be searched

Comment (ringabout) [reply changelogs/changelog_2_0_0.md:263]: Yeah, make sense, I will correct the changelog.

**Pull request merged**